hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,290 @@ title: "OTP Controller Technical Specification"
+
+### Software Configuration Partitions
+
+The software configuration partitions are used as non-volatile storage for flags, configuration and calibration data.
+As such, the contents of this partition are usually consumed once as part of code execution, or moved to another storage compartment somewhere in the design.
+For example, the clock calibration values and the LDO calibration values are programmed to the analog sensor top (AST) at startup.
+
+As such, it is not necessary to check periodically at the OTP source.
+Instead, software can simply check as part of secure boot and take other measures when these values are programmed into peripherals.
+
+For this partition it is thus the responsibility of software to calculate the integrity digest and program it into the OTP.
+It is also reasonable to shadow (parts of) this partition in main memory, and there is not an immediate impact from OTP contents to hardware.
+
+### Hardware Configuration and Secret Partitions
+
+The hardware and secret partitions directly affect downstream hardware.
+The contents must go through periodic integrity checks and therefore the stored digest is calculated by hardware when software provides the intent to lock (as opposed to the software partitions where the digest has to be calculated by software).
+
+### Life Cycle Partition
+
+The life cycle partition cannot be locked and will therefore not contain a stored digest.
+
+## Secret vs Non-Secret Partitions
+
+Non-secret OTP partitions hold data that can be public; or data that has no impact on security.
+For example, the current value of lock bits, life cycle state or clock calibration value.
+These values are stored in OTP as plaintext.
+
+Secret partitions contain data that are critical to security, for example flash scrambling keys, device root secret and unlock tokens.
+These values are stored scrambled in OTP, and are descrambled upon read.
+The currently employed cipher is PRESENT, as it lends itself well to iterative decomposition, and it is a proven lightweight block cipher (see also [PRESENT Scrambling Primitive]({{< relref "hw/ip/prim/doc/prim_present.md" >}}).
+The usage of a block cipher however implies that the secret partitions can only be written in 64bit chunks.
+
+Further, the contents of a particular secret partition are not readable by software once locked (other than the digest which must be always readable); while non-secret partitions are always readable unless read accessibility is explicitly removed by software.
+
+Unfortunately, secret partitions must utilize a global netlist key for the scrambling operation, as there is no other non-volatile storage to store a unique key.
+
+
+## Partition Checks
+
+### Integrity
+
+Once the appropriate partitions have been locked, the hardware integrity checker employs two integrity checks to verifies the content of the volatile buffer registers:
+
+1. All buffered partitions have additional Byte parity protection that is concurrently monitored.
```

Comment:
Is there a reason `Byte` is capitalized?

Reply:
Hm... I guess I do that out of habit since Byte is also capitalized when abbreviated (B), but that may not be entirely correct. let me fix those.